### ISSUE TITLE
`clj-commons/fs` changes standard clojure behaviour in `clojure.java.io/file`

### ISSUE THREAD

**Issue by ikappaki** [clojure-emacs/refactor-nrepl#355]
## Expected behavior

The REPL shouldn't change standard Clojure behaviour, otherwise it might lead to unexpected behaviour in release builds.

Brining up a REPL with clj-refactor installed allows `clojure.java.io/file` to support `java.nio.file.Path` as an argument, while it does not in the standard core, .e.g. if we bring up a vanilla REPL (wo clj-refactor) and try to open a `Path` with `io/file` it fails as expected:
```clojure
user> (-> "a-path" (java.nio.file.Paths/get (into-array String [])) clojure.java.io/file)
Execution error (IllegalArgumentException) at user/eval5725 (REPL:59).
No implementation of method: :as-file of protocol: #'clojure.java.io/Coercions found for class: sun.nio.fs.WindowsPath
```
Analysis follows at the end.

## Actual behavior

clj-refactor changes standard Clojure behaviour with regards to opening files with `clojure.java.io/file`.

## Steps to reproduce the problem

1. `M-x cider-jack-in` to a minimal `deps.edn` file.
2. At the REPL, try to open a file from a `java.nio.Path`
```clojure
user> (-> "a-path" (java.nio.file.Paths/get (into-array String [])) clojure.java.io/file)
#object[java.io.File 0x62eac79 "a-path"]
```
i.e. it returns the file from the `PATH`, while this is unsupported in standard Clojure behaviour.

### clj-refactor.el version information

clj-refactor 3.2.2 (package: 20211117.1008), refactor-nrepl 3.1.0

### CIDER version information

*Include here the version string displayed when
CIDER's REPL is launched. Here's an example:*

```
;; CIDER 1.2.0snapshot (package: 20211126.1158), nREPL 0.9.0-beta5
;; Clojure 1.10.3, Java 17
```

### Emacs version

GNU Emacs 27.2 

### Operating system

MS-Windows 10

## Analysis

The standard supported `io/file` arguments types are defined by the `io/Coercions` protocol's `as-file` method and [are `String`, `File`, `URL` and `URI`.](https://github.com/clojure/clojure/blob/abe19832c0294fec4c9c55430c9262c4b6d2f8b1/src/clj/clojure/java/io.clj#L44)

The support for `Path`'s in `io/file` is brought in by the  [clj-common/fs](https://github.com/clj-commons/fs.git) dependency  adding [`Path` to `io/Coercions`](https://github.com/clj-commons/fs/blob/04c7b1abcc2f3cdff782dae7f022781aed11c42a/src/me/raynes/fs.clj#L144). `clj-common/fs` is part of `refactor-nrepl` middleware injected by `clj-refactor`:

```clojure
user> (-> "a-path" (java.nio.file.Paths/get (into-array String [])) clojure.java.io/file)
#object[java.io.File 0x17537a12 "a-path"]
user> (:impls clojure.java.io/Coercions)
{nil
 {:as-file #function[clojure.java.io/fn--11402],
  :as-url #function[clojure.java.io/fn--11404]},
 java.lang.String
 {:as-file #function[clojure.java.io/fn--11406],
  :as-url #function[clojure.java.io/fn--11408]},
 java.io.File
 {:as-file #function[clojure.java.io/fn--11410],
  :as-url #function[clojure.java.io/fn--11412]},
 java.net.URL
 {:as-url #function[clojure.java.io/fn--11414],
  :as-file #function[clojure.java.io/fn--11416]},
 java.net.URI
 {:as-url #function[clojure.java.io/fn--11418],
  :as-file #function[clojure.java.io/fn--11420]},
 java.nio.file.Path
 {:as-file
  #function[refactor-nrepl.inlined-deps.fs.v1v6v307.me.raynes.fs/eval2296/fn--2297],
  :as-url
  #function[refactor-nrepl.inlined-deps.fs.v1v6v307.me.raynes.fs/eval2296/fn--2299]}}
```

I was caught off guard with this while providing a PR about filtering filenames given a path and would like to bring this to your attention. Thinking my input path was a `String` I was passing it `io/file` with the aim to retrieve its filename with `.getName`, but it turned out it was a `Path`. It should have failed while I was testing on the REPL before reaching the release build, but it got missed because of this. I even naively thought afterwards that `Path`d were supported in the standard API, but it does not https://clojure.atlassian.net/browse/CLJ-2333.

IMHO REPL middleware shouldn't modify standard behaviour since this might lead to confusion with what we are getting during dev vs release. Not exactly sure how to fix this particular issue since this is not part of the middleware core but rather an external dependency, but wanted to open the discussion anyway. 

https://github.com/nextjournal/clerk/pull/28

Thanks!

**Comment by vemv** [comment]
Created https://github.com/clj-commons/fs/pull/11

Transfered the issue to refactor-nrepl as well

**Comment by ikappaki** [comment]
> If there's a specific problem, feel free to open it as an issue (if it's _this_ one i.e. the `fs` matter, please rest assured that it will be fixed)

I think we've exhausted the subject, no need for raising another issue. Thanks for looking at the fs 👍🏻

**Comment by vemv** [comment]
To be released soon, maybe I'll bundle something else while we're there

**Comment by vemv** [comment]
Released as 3.2.0
